src/main/java/net/neoforged/neoforge/common/extensions/ValueInputExtension.java

```diff
@@ -25,4 +26,25 @@ default Set<String> keySet() {
         //noinspection deprecation
         return self().read(MapCodec.assumeMapUnsafe(CompoundTag.CODEC)).orElseThrow().keySet();
     }
+
+    /**
+     * Read the given {@code child} object from the given {@code key}, with the possibility that the {@code child} object may be empty.
+     *
+     * @param key   the key to read the child from
+     * @param child the child to read from given key
+     */
+    default void readChildOrEmpty(String key, ValueIOSerializable child) {
+        child.deserialize(self().childOrEmpty(key));
+    }
```

Comment:
I've thought about it more, and I see no reason for this method. `ValueIOSerializable` is implemented on an object to handle reads and writes. If the read is never called, the object keeps its default state. So, given that, I don't see the purpose of having it go through an empty input if its just going to replace the fields with the same values.